chronos/asyncfutures2.nim

```diff
@@ -561,8 +547,20 @@ proc asyncSpawn*(future: Future[void]) =
     elif future.cancelled():
       raise newException(FutureDefect, getCancelMessage())
 
-proc asyncDiscard*[T](future: Future[T]) {.deprecated.} = discard
-  ## This is async workaround for discard ``Future[T]``.
+proc asyncCheck*[T](future: Future[T]) {.
+    deprecated: "Raises Defect on future failure, fix your code and use asyncSpawn!".} =
+  ## This function used to raise an exception through the `poll` call if
+  ## the given future failed - there's no way to handle such exceptions so this
+  ## function is now an alias for `asyncSpawn`
+  ##
+  asyncSpawn(future)
```

Comment:
Won't there be some `Future[void]` versus `Future[T]` incompatiblities here? 
I noticed this error at least in nim-eth's (old) kademlia when testing by running the p2p tests.

Reply:
yes, I have a fix locally for that which discards the T

Reply:
pushed